org.eclipse.buildship.ui.test/src/main/groovy/org/eclipse/buildship/ui/view/execution/OpenBuildScanActionTest.groovy

```diff
@@ -0,0 +1,196 @@
+        setup:
+        File projectDir = dir('project-with-fake-build-scan') {
+            file 'build.gradle', '''
+                task publishFakeBuildScanA {
+                    doLast {
+                        println 'Publishing build information...'
+                        println 'https://scans.gradle.com/s/A'
+                    }
+                }
+                task publishFakeBuildScanB {
+                    doLast {
+                        println 'Publishing build information...'
+                        println 'https://scans.gradle.com/s/B'
+                    }
+                }
+            '''
+        }
+
+        when:
+        synchronizeAndWait(projectDir)
+        launchTaskAndWait(projectDir, 'publishFakeBuildScanA')
+        launchTaskAndWait(projectDir, 'publishFakeBuildScanB')
+
+        then:
+        view.pages.size() == 2
+        view.pages[0].openBuildScanAction.enabled
+        view.pages[0].openBuildScanAction.buildScanUrl == 'https://scans.gradle.com/s/A'
+        view.pages[1].openBuildScanAction.enabled
+        view.pages[1].openBuildScanAction.buildScanUrl == 'https://scans.gradle.com/s/B'
+    }
+
+    @Ignore
+    def "Build scan publishes real build scan"(String gradleVersion, String buildScanVersion, List<String> arguments) {
+        setup:
+        File projectDir = dir('project-with-build-scan') {
+            file 'build.gradle', """
+                buildscript {
+                    repositories { maven { url 'https://plugins.gradle.org/m2/' } }
+                    dependencies { classpath 'com.gradle:build-scan-plugin:$buildScanVersion' }
+                }
+                apply plugin: 'com.gradle.build-scan'
+                buildScan {
+                    licenseAgreementUrl = 'https://gradle.com/terms-of-service'
+                    licenseAgree = 'yes'
+                }
+                task foo { doLast { println 'bar' } }
+            """
+        }
+
+        when:
+        importAndWait(projectDir, GradleDistribution.forVersion(gradleVersion))
+        launchTaskAndWait(projectDir, 'foo', arguments)
+
+        then:
+        view.pages.size() == 1
+        view.pages[0].openBuildScanAction.enabled
+        view.pages[0].openBuildScanAction.buildScanUrl.startsWith 'https://'
+
+        where:
+        gradleVersion | buildScanVersion | arguments
+        '3.5'         | '1.7.1'          | ['--scan']
+        '3.3'         | '1.6'            | ['-Dscan']
+    }
+
+    private void launchTaskAndWait(File projectDir, String task, List<String> arguments = []) {
+        new RunGradleBuildLaunchRequestJob(createLaunch(task, projectDir, arguments)).schedule()
+        waitForGradleJobsToFinish()
+        waitForBuildFinishedMessage()
+    }
+
+    private void waitForBuildFinishedMessage() {
+        waitFor { consoleListener.activeConsole.document.get().contains('BUILD SUCCESSFUL') }
+    }
+
+    private void waitFor(int timeout = 5000, Closure condition) {
```

Comment:
Unfortunately not. The console output changes asynchronously, so the build job finishes before the build success message is printed.